gammapy/modeling/models/core.py

```diff
@@ -815,6 +816,97 @@ def to_template_sky_model(self, geom, spectral_model=None, name=None):
             spectral_model=spectral_model, spatial_model=spatial_model, name=name
         )
 
+    def plot_regions(self, ax=None, **kwargs):
+        """ Plot extent of the spatial models on a given wcs axis
+
+        Parameters
+        ----------
+        ax : `~astropy.vizualisation.WCSAxes`
+            Axes to plot on. If no axes are given, an all-sky wcs
+            is chosen using a CAR projection
+        **kwargs : dict
+            Keyword arguments passed to `~matplotlib.axes.Axes.add_artist`
+
+
+        Returns
+        -------
+        ax : `~astropy.vizualisation.WcsAxes
+        """
+        import matplotlib.pyplot as plt
+        from astropy.visualization.wcsaxes import WCSAxes
+        from gammapy.modeling.models import PointSpatialModel
+
+        if ax is None or not isinstance(ax, WCSAxes):
+            wcs = WcsGeom.create().wcs
+            ax = plt.gca(projection=wcs)
```

Comment:
Have you checked this minimal example:
```python3
from gammapy.maps import RegionGeom
from gammapy.utils.regions import make_region
import numpy as np

regions = []

lons = np.random.uniform(-5, 5, 100)
lats = np.random.uniform(-3, 3, 100)
radii = np.random.uniform(0.1, 0.7, 100)

for lon, lat, radius in zip(lons, lats, radii):
    region = make_region(f"icrs;circle({lon}, {lat}, {radius})")
    regions.append(region)

geom = RegionGeom.from_regions(regions)
ax = geom.plot_region()
```
In general Gammapy can determine the minimal contained WCS for an arbitrary list of regions...so this allows you to  possibly choose a better selected ROI for the models...

Reply:
Then I can add a `Models.to_spatial_region()` which returns a list of regions and use this inside the plotting code. Is that OK?